examples/src/main/java/io/grpc/examples/multiplex/SharingClient.java

```diff
@@ -0,0 +1,248 @@
+   * greeting. The second argument is the target server.
+   * You can see the multiplexing in the server logs.
+   */
+  public static void main(String[] args) throws Exception {
+    String user = "world";
+    // Access a service running on the local machine on port 50051
+    String target = "localhost:50051";
+    // Allow passing in the user and target strings as command line arguments
+    if (args.length > 0) {
+      if ("--help".equals(args[0])) {
+        System.err.println("Usage: [name [target]]");
+        System.err.println("");
+        System.err.println("  name    The name you wish to be greeted by. Defaults to " + user);
+        System.err.println("  target  The server to connect to. Defaults to " + target);
+        System.exit(1);
+      }
+      user = args[0];
+    }
+    if (args.length > 1) {
+      target = args[1];
+    }
+
+    // Create a communication channel to the server, known as a Channel. Channels are thread-safe
+    // and reusable. It is common to create channels at the beginning of your application and reuse
+    // them until the application shuts down.
+    //
+    // For the example we use plaintext insecure credentials to avoid needing TLS certificates. To
+    // use TLS, use TlsChannelCredentials instead.
+    ManagedChannel channel = Grpc.newChannelBuilder(target, InsecureChannelCredentials.create())
+        .build();
+    List<String> echoInput = ImmutableList.of("some", "thing", "wicked", "this", "way", "comes");
+    try {
+      SharingClient client = new SharingClient(channel);
+
+      StreamingFuture<List<String>> future = client.initiateEchos(echoInput);
+      client.greet1(user + " the great");
+      client.greet2(user + " the lesser");
+      client.greet1(user + " the humble");
+      // Receiving happens asynchronously
+
+      String resultStr = future.get(1, TimeUnit.MINUTES).toString();
+      System.out.println("The echo requests and results were:");
+      System.out.println(echoInput.toString());
+      System.out.println(resultStr);
+
+      if (!future.isDone()) {
+        System.err.println("Streaming rpc failed to complete in 1 minute");
+      }
+    } finally {
+      // ManagedChannels use resources like threads and TCP connections. To prevent leaking these
+      // resources the channel should be shut down when it will no longer be used. If it may be used
+      // again leave it running.
+      channel.shutdownNow().awaitTermination(5, TimeUnit.SECONDS);
+    }
+  }
+
+  private class StreamingFuture<RespT> extends AbstractFuture<RespT> {
+    private StreamObserver<EchoResponse> requestObserver = null;
+
+    private void setObserver(StreamObserver<EchoResponse> requestObserver) {
+      this.requestObserver = requestObserver;
+    }
+
+    @Override
+    protected void interruptTask() {
+      if (requestObserver != null) {
+        requestObserver.onError(new RuntimeException("Stream was cancelled through a future"));
```

Comment:
error message is not straightforward - the code didn't cancel anywhere.

Reply:
Redone to use Status.ABORTED